lib/core/sync_job_runner.rb

```diff
@@ -52,14 +51,18 @@ def do_sync!
       begin
         Utility::Logger.debug("Successfully claimed job for connector #{@connector_settings.id}.")
 
-        @connector_instance.do_health_check!
+        # will be replaced when job claiming returns full job description object
+        job_description = { :filtering => @connector_settings.filtering }
+        connector_instance = Connectors::REGISTRY.connector(@connector_settings.service_type, @connector_settings.configuration, job_description)
+
+        connector_instance.do_health_check!
 
         incoming_ids = []
         existing_ids = ElasticConnectorActions.fetch_document_ids(@connector_settings.index_name)
 
         Utility::Logger.debug("#{existing_ids.size} documents are present in index #{@connector_settings.index_name}.")
 
-        @connector_instance.yield_documents do |document|
+        connector_instance.yield_documents(job_description) do |document|
```

Comment:
```suggestion
        connector_instance.yield_documents do |document|
```

job description now passed via the constructor